src/test/ui/parser/pat-lt-bracket-7.rs

```diff
@@ -1,3 +1,5 @@
 fn main() {
     for thing(x[]) in foo {} //~ ERROR: expected one of `)`, `,`, or `@`, found `[`
+    //~^ ERROR cannot find value `foo` in this scope
+    //~| ERROR cannot find tuple struct/variant `thing` in this scope
```

Comment:
More off-topic errors.

Reply:
Addressed the off-topic errors, I hope... I introduced `const RECOVERY_WITNESS...` to check that error recovery actually happens.